packages/xo-server/src/xapi-stats.js

```diff
@@ -411,40 +202,57 @@ export default class XapiStats {
       .then(response => response.readAll().then(JSON5.parse))
   }
 
-  async _getLastTimestamp (xapi, host, step) {
-    if (this._hosts[host.address][step] === undefined) {
-      const serverTimeStamp = await getServerTimestamp(xapi, host)
-      return serverTimeStamp - step * RRD_POINTS_PER_STEP[step] + step
-    }
+  async _getValidTimestamp (xapi, host, step) {
```

Comment:
I still don't understand what it does…

Reply:
The function returns a timestamp that it will be used to get data recorded after this timestamp.
_getValidTimestamp -> _getTimestampOfBeginningRecording

Reply:
`_getNextTimestamp`?